trulens/nn/attribution.py

```diff
@@ -245,101 +270,114 @@ class given by the first integer against the class given by the
 
     def attributions(
         self, *model_args: ArgsLike, **model_kwargs: KwargsLike
-    ) -> np.ndarray:
+    ) -> Union[DataLike, ArgsLike[DataLike], ArgsLike[ArgsLike[DataLike]]]:
+        # NOTE: not symbolic
+
+        B = get_backend()
+
+        if isinstance(
+                model_args,
+                DATA_CONTAINER_TYPE) and len(model_args) == 1 and isinstance(
+                    model_args[0], DATA_CONTAINER_TYPE):
+            model_args = model_args[0]
+
         model_inputs = ModelInputs(model_args, model_kwargs)
 
-        doi_cut = self.doi.cut() if self.doi.cut() else InputCut()
+        # Will cast results to this data container type.
+        return_type = type(model_inputs.first())
 
-        doi_val = self.model.fprop(
-            model_args=model_inputs.args,
-            model_kwargs=model_inputs.kwargs,
-            to_cut=doi_cut
-        )
+        # Create a message for out-of-memory errors regarding float and batch size.
+        if len(list(model_inputs.values())) == 0:
+            batch_size = 1
+        else:
+            batch_size = model_inputs.first().shape[0]
 
-        # DoI supports tensor or list of tensor. unwrap args to perform DoI on
-        # top level list
+        param_msgs = [
+            f"float size = {B.floatX_size} ({B.floatX}); consider changing to a smaller type.",
+            f"batch size = {batch_size}; consider reducing the size of the batch you send to the attributions method."
```

Comment:
nit: use the parameter name so they know which parameter is being referred to

Reply:
For the ones here, I don't think there is a consistent parameter name they could use to adjust things. They have to look up how, for example, batch sizing is done in whatever system they are using.